test/unit/modules/accounts.js

```diff
@@ -0,0 +1,725 @@
+var crypto = require('crypto');
+var async = require('async');
+var sinon = require('sinon');
+
+var chai = require('chai');
+var expect = require('chai').expect;
+var constants = require('../../../helpers/constants.js');
+var _  = require('lodash');
+
+var Rounds = require('../../../modules/rounds.js');
+var AccountLogic = require('../../../logic/account.js');
+var AccountModule = require('../../../modules/accounts.js');
+var TransactionLogic = require('../../../logic/transaction.js');
+var TransactionModule = require('../../../modules/transactions.js');
+var DelegateModule = require('../../../modules/delegates.js');
+var BlocksModule = require('../../../modules/blocks.js');
+var modulesLoader = require('../../common/initModule').modulesLoader;
+
+
+var accountSecret = 'actress route auction pudding shiver crater forum liquid blouse imitate seven front';
+var validAccount = {
+	username: 'genesis_100',
+	isDelegate: 1,
+	u_isDelegate: 1,
+	secondSignature: 0,
+	u_secondSignature: 0,
+	u_username: 'genesis_100',
+	address: '10881167371402274308L',
+	publicKey: 'addb0e15a44b0fdc6ff291be28d8c98f5551d0cd9218d749e30ddb87c6e31ca9',
+	secondPublicKey: null,
+	vote: '9820020609280331',
+	rate: '0',
+	delegates: null,
+	u_delegates: null,
+	multisignatures: null,
+	u_multisignatures: null,
+	multimin: 0,
+	u_multimin: 0,
+	multilifetime: 0,
+	u_multilifetime: 0,
+	blockId: '10352824351134264746',
+	nameexist: 0,
+	u_nameexist: 0,
+	producedblocks: 27,
+	missedblocks: 1,
+	fees: '231386135',
+	rewards: '0',
+	virgin: 1
+};
+
+describe('account', function () {
+
+	var account; 
+	var accountLogic;
+	var accountModuleDependencies;
+
+	before(function (done) {
+		async.auto({
+			rounds: function (cb) {
+				modulesLoader.initModule(Rounds, modulesLoader.scope,cb);
+			},
+			accountLogic: function (cb) {
+				modulesLoader.initLogicWithDb(AccountLogic, cb);
+			},
+			blockModule: ['accountLogic', function (result, cb) {
+				modulesLoader.initModuleWithDb(BlocksModule, cb, {
+					logic : { /* dependencies not included */},
+				});
+			}],
+			transactionLogic: ['accountLogic', function (result, cb) {
+				modulesLoader.initLogicWithDb(TransactionLogic, cb, {
+					account: result.accountLogic
+				});
+			}],
+			delegateModule: ['transactionLogic', function (result, cb) {
+				modulesLoader.initModuleWithDb(DelegateModule, cb, {
+					logic: {
+						transaction: result.transactionLogic
+					}
+				});
+			}],
+			transactionModule: ['transactionLogic', function (result, cb) {
+				modulesLoader.initModuleWithDb(TransactionModule, cb, {
+					transaction: result.transactionLogic
+				});
+			}]
+		}, function (err, result) {
+			modulesLoader.initModuleWithDb(AccountModule, function (err, __accountModule) {
+				expect(err).to.not.exist;
+
+				account = __accountModule;
+				accountLogic = result.accountLogic;
+
+				// for correctly initializing setting blocks module
+				result.blockModule.lastBlock.set({height: 10});
```

Comment:
This is a dummy value I am setting so that we fake-load the modules correctly. The last block value isn't used to query the DB, we only use it [here](https://github.com/SargeKhan/lisk/blob/5037cc32629a0e7809b5b090b1cb29058fd7a55f/modules/delegates.js#L401) for running tests for `Accounts.shared#getDelegates` (which internally calls delegates module) function.